### PR TITLE
[Memory] reuse image cache for asset to reduce memory usage, alternative for #1590

### PR DESCRIPTION
**Pull request by jianjunwoo**
As there are several image provider implementation classes, I create extension func to enable cache.
There is another way to implement the cache mechanism compared to PR https://github.com/airbnb/lottie-ios/pull/1590

## Change summary:
Before, the lottie animation create an image for each layer from asset. 
In this PR, the `BundleImageProvider ` is enhanced with image cache to reduce memory usage when creating layers with same asset.

## Please review:

|| before | after |
|-|-|-|
| Sample Lottie |before | after|

### CODE REVIEW

File: Sources/Public/iOS/CachedImageProvider.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  // MARK: Internal
+
+  let imageCache: NSCache<NSString, CGImage> = .init()
+  let imageProvider: AnimationImageProvider
+}
+
+extension AnimationImageProvider {
+  public var cachedImageProvider: AnimationImageProvider {
```
[the remark this answers is not in the document]
Comment (calda) [reply Sources/Public/iOS/CachedImageProvider.swift:43]: What use cases do you have in mind for this extension, given that the built-in `BundleImageProvider` has its own cache as-of #1590?

File: Sources/Public/Animation/AnimationViewInitializers.swift [hunk 2]
```diff
@@ -26,7 +26,7 @@ extension AnimationView {
     animationCache: AnimationCacheProvider? = LRUAnimationCache.sharedCache)
   {
     let animation = Animation.named(name, bundle: bundle, subdirectory: nil, animationCache: animationCache)
-    let provider = imageProvider ?? BundleImageProvider(bundle: bundle, searchPath: nil)
+    let provider = imageProvider ?? BundleImageProvider(bundle: bundle, searchPath: nil).cachedImageProvider
```
Comment (calda) [inline Sources/Public/Animation/AnimationViewInitializers.swift:29]: If `BundleImageProvider` has its own internal cache, does it need to be wrapped in the cache wrapper? Does that mean that the image will be cached in two different places?

File: Sources/Public/iOS/CachedImageProvider.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcelofabri) [reply Sources/Public/iOS/CachedImageProvider.swift:43]: I think this PR is just an alternative implementation of #1590

[the remark this answers is not in the document]
Comment (jianjunwoo) [reply Sources/Public/iOS/CachedImageProvider.swift:43]: yeah, this is an alternative implementation of https://github.com/airbnb/lottie-ios/pull/1590

[the remark this answers is not in the document]
Comment (jianjunwoo) [reply Sources/Public/iOS/CachedImageProvider.swift:43]: > What use cases do you have in mind for this extension, given that the built-in `BundleImageProvider` has its own cache as-of #1590?

`FilepathImageProvider` is another implementation of AnimationImageProvider. With this extension, we could easily extent the cache capability to it. @calda

File: Sources/Public/Animation/AnimationViewInitializers.swift (hunk 2 above)
Comment (jianjunwoo) [reply Sources/Public/Animation/AnimationViewInitializers.swift:29]: It's an alternative implementation of https://github.com/airbnb/lottie-ios/pull/1590, and would like to make it easy to extent the cache capability for other image providers.

**Pull request closed without merging**